flang/CMakeLists.txt

```diff
@@ -473,6 +473,18 @@ if (APPLE)
   endif()
 endif()
 
+# Set up job pools for flang.  Some of the flang sources take a lot of memory to
+# compile, so allow users to limit the number of parallel flang jobs.  This is
+# useful for building flang alongside several other projects since you can use
+# the maximum number of build jobs for the other projects while limiting the
+# number of flang compile jobs.
+#
+# We want this set to infinity by default
+set(FLANG_PARALLEL_COMPILE_JOBS 0 CACHE STRING
+  "The maximum number of concurrent compilation jobs (Ninja only)")
```

Comment:
There was a comment by @Meinersbur that this default is not a good setting.

https://github.com/llvm/llvm-project/issues/112789#issuecomment-2429553410

Separately, why doesn't this apply to Make?

Reply:
Documented [here](https://cmake.org/cmake/help/latest/prop_gbl/JOB_POOLS.html):

> Ninja only: List of available pools.

Ninja has job pools, GNU Make does not.